addons/mail/static/src/models/thread/thread.js

```diff
@@ -1192,6 +1262,26 @@ function factory(dependencies) {
             return [['replace', this.messages.filter(message => message.isNeedaction)]];
         }
 
+        /**
+         * @returns {integer|undefined}
+         * @private
+         */
+        _computeMessageBeforeNewMessageSeparatorId() {
+            if (this.model !== 'mail.channel') {
+                return clear();
+            }
+            if (this.localMessageUnreadCounter === 0) {
+                return clear();
+            }
+            if (!this.lastSeenByCurrentPartnerMessageId) {
+                return this.lastTransientMessage ? this.lastTransientMessage.id : 0;
+            }
```

Comment:
If the user has seen no message at all, but some real unread messages exist, the value should be 0 or "last transient that is before the first real message" (not last transient alone, because that could be at the end of the list).

Reply:
This seems not in accordance with your rule "But if transient messages appear somehow after the last seen message in the thread, they should be considered as unseen and the counter adapted accordingly."

Let's consider this situation :
```
lastSeenMessage is false.
[message 1]
[message 2]
[message 3]
[message 4]
[message 5 (transient)]
[message 6 (transient)]
```

Transient messages should be counted as unseen so counter will be 6 and so the separator should still be in 0.
Or here, "last transient that is before the first real message" does not exist.
In the following situation, it's the same I think but "last transient that is before the first real message" is message 1 but then, this would show the separator above 5 messages while unread counter is 6 ..
```
lastSeenMessage is false.
[message 1 (transient) ]
[message 2]
[message 3]
[message 4]
[message 5 (transient)]
[message 6 (transient)]
```

Reply:
I guess in this case it would be acceptable if the separator was at the top, before the transient message, and have the counter be 6. Especially if that allows having one less strange condition.

Reply:
But the ideal actually would be to have a proper local "last seen id" that can include transient and that is just not sync with the server?

Reply:
If we have that we can just keep one set of rules, transient or not.